connect/runtime/src/test/java/org/apache/kafka/connect/integration/RebalanceSourceConnectorsIntegrationTest.java

```diff
@@ -176,10 +184,13 @@ public void testReconfigConnector() throws Exception {
                    + "within " + CONNECTOR_SETUP_DURATION_MS + "ms",
                 restartLatch.await(CONNECTOR_SETUP_DURATION_MS, TimeUnit.MILLISECONDS));
 
-        // And wait for the Connect to show the connectors and tasks are running
+        // And wait for Connect to show the connectors and tasks are running
         waitForCondition(() -> this.assertConnectorAndTasksRunning(CONNECTOR_NAME, NUM_TASKS).orElse(false),
                 CONNECTOR_SETUP_DURATION_MS, "Connector tasks did not start in time.");
 
+        // wait for the connector's tasks to have produced records
+        connectorHandle.awaitRecords(anotherTopic, recordWriteDurationMs);
+
```

Comment:
Part 3b: Wait up to 60 seconds for the connector to actually write 100 records to the new topic.